internal/doctor/core.go

```diff
@@ -0,0 +1,142 @@
+package doctor
+
+import (
+	"github.com/gin-gonic/gin"
+	"hospitalApp/internal/patient"
+	"log"
+	"net/http"
+	"time"
+)
+
+type CreateRequestDoctorData struct {
+	DoctorId  uint   `json:"doctor_id"`
+	Name      string `json:"name"`
+	ContactNo string `json:"contact_no"`
+}
+
+type UpdateRequestDoctorData struct {
+	DoctorId  uint   `json:"doctor_id"`
+	Name      string `json:"name"`
+	ContactNo string `json:"contact_no"`
+}
+
+type UpdateRequestPatientData struct {
+	DoctorId uint            `json:"doctor_id"`
+	Patient  patient.Patient `json:"patient"`
+}
+
+func GetEntityByName(context *gin.Context) {
+	name := context.Query("name")
+	doctor, err := FindUserByName(name)
+	log.Printf("Doctor data get by Name(%q) : %+v", name, doctor)
+	if err != nil {
+		context.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+	}
+
+	if doctor.DoctorId == 0 {
+		context.JSON(http.StatusOK, gin.H{"data": ""})
+		return
```

Comment:
how are you making the ID's. Are you getting alphanumeric 5 digits? Also how are you validating it

Reply:
Added generateKey function [here](https://github.com/nikhilsharma1100/hospitalApp/pull/1/files#diff-cdf7e8691d355860ce1a5b84d3f4d6b701b931eefc370c2abda1d968e1ef0d2aR158)